stdlib/Distributed/src/remotecall.jl

```diff
@@ -548,14 +548,35 @@ Further calls to `fetch` on the same reference return the cached value. If the r
 is an exception, throws a [`RemoteException`](@ref) which captures the remote exception and backtrace.
 """
 function fetch(r::Future)
-    r.v !== nothing && return something(r.v)
-    v = call_on_owner(fetch_ref, r)
-    r.v = Some(v)
+    v = @atomic :sequentially_consistent r.v
```

Comment:
For a pointer, `@atomic :sequentially_consistent r.v` is usually indistinguishable from plain `r.v` (which is a consume load), so this seems odd to mark explicitly

Reply:
Do you mean a relaxed load? Nobody knows how to define consume for the C/C++ abstract machine IIUC.

Reply:
The linux kernel uses consume loads (https://www.kernel.org/doc/Documentation/RCU/rcu_dereference.txt), and they are well-supported by most hardware (and I think Java mandates them too?). They are currently only an issue for the abstract machine.

Reply:
Yeah, but most people don't write or control assembly. They program against an abstract machine rather than concrete hardware. I think consume ordering makes sense to you because you understand how a given piece of Julia code is translated to the assembly by a complex beast like Julia+LLVM. Most people can't do that.